e2e/app/page/dataset-build-page.ts

```diff
@@ -0,0 +1,99 @@
+import Table from 'app/component/table';
+import Button from 'app/element/button';
+import Checkbox from 'app/element/checkbox';
+import {waitWhileLoading} from 'utils/test-utils';
+import {waitForDocumentTitle} from 'utils/waits-utils';
+import {xPathOptionToXpath} from 'app/element/xpath-defaults';
+import {ElementType} from 'app/xpath-options';
+import AuthenticatedPage from './authenticated-page';
+import CohortBuildPage from './cohort-build-page';
+
+const PageTitle = 'Dataset Page';
+
+enum LabelAlias {
+  SelectValues = 'Select Values',
+  SelectConceptSets = 'Select Concept Sets',
+  SelectCohorts = 'Select Cohorts',
+}
+
+export default class DatasetBuildPage extends AuthenticatedPage {
+
+  async isLoaded(): Promise<boolean> {
+    try {
+      await Promise.all([
+        waitForDocumentTitle(this.page, PageTitle),
+        waitWhileLoading(this.page),
+      ]);
+      return true;
+    } catch (e) {
+      console.log(`DatasetBuildPage isLoaded() encountered ${e}`);
+      return false;
+    }
+  }
+
+  async getAddCohortsButton(): Promise<Button> {
+    const buttonXpath = xPathOptionToXpath({name: LabelAlias.SelectCohorts, ancestorLevel: 3, iconShape: 'is-solid'});
+    return new Button(this.page, buttonXpath);
+  }
+
+  async clickAddCohortsButton(): Promise<CohortBuildPage> {
+    const addCohortButton = await this.getAddCohortsButton();
+    await addCohortButton.clickAndWait();
+    const cohortPage = new CohortBuildPage(this.page);
+    return cohortPage.waitForLoad();
+  }
+
+  async selectCohorts(cohortNames: string[]): Promise<void> {
+    for (const cohortName of cohortNames) {
+      const xpath = this.getCheckboxXpath(LabelAlias.SelectCohorts, cohortName);
+      const checkbox = new Checkbox(this.page, xpath);
+      await checkbox.check();
+    }
+  }
+
+  async selectConceptSets(conceptSetNames: string[]): Promise<void> {
+    for (const conceptSetName of conceptSetNames) {
+      const xpath = this.getCheckboxXpath(LabelAlias.SelectConceptSets, conceptSetName);
+      const checkbox = new Checkbox(this.page, xpath);
+      await checkbox.check();
+      await waitWhileLoading(this.page);
+    }
+  }
+
+  async selectValues(selectAll: boolean = false, values?: string[]): Promise<void> {
```

Comment:
nit: I don't think the boolean type should be necessary in this situation; type I believe should just be implicit based ont he default value